Improve MySQL Compatibility by supporting SCHEMA()

dveeden opened the pull request:
Basically `DATABASE()` and `SCHEMA()` are aliases in MySQL.

This helps with allowing MySQL Shell to get one step closer to a functioning connection.

To see what MySQL Shell does:

```
mysqlsh --verbose=4 --log-sql=unfiltered mysql://default@127.0.0.1:9004
```

This might also bring other tools a step closer to working with the MySQL compatibility of ClickHouse.

Ref: #9336

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):

- Improvement



### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
`SCHEMA()` was added as alias for `DATABASE()` to improve MySQL compatibility.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

<!---
Directly edit documentation source files in the "docs" folder with the same pull-request as code changes

or

Add a user-readable short description of the changes that should be added to docs.clickhouse.com below.

At a minimum, the following information should be added (but add more as needed).
- Motivation: Why is this function, table engine, etc. useful to ClickHouse users?

- Parameters: If the feature being added takes arguments, options or is influenced by settings, please list them below with a brief explanation.

- Example use: A query or command.
-->


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

dveeden committed:
Improve MySQL Compatibility by supporting SCHEMA()

Basically `DATABASE()` and `SCHEMA()` are aliases in MySQL.

This helps with allowing MySQL Shell to get one step closer to a
functioning connection.

To see what MySQL Shell does:

```
mysqlsh --verbose=4 --log-sql=unfiltered mysql://default@127.0.0.1:9004
```

This might also bring other tools a step closer to working with the
MySQL compatibility of ClickHouse.

Ref: #9336

Signed-off-by: Daniël van Eeden <git@myname.nl>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=53587) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/ClickHouse/ClickHouse?pullRequest=53587) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ClickHouse/ClickHouse?pullRequest=53587) it.</sub>

robot-clickhouse-ci-1 commented:
<!-- automatic status comment for PR #53587 from dveeden/ClickHouse:mysql_schema -->
This is an automated comment for commit 212686c5c4f349ffc3adff812c6783fecc6a18a9 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/53587/212686c5c4f349ffc3adff812c6783fecc6a18a9/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

alexey-milovidov commented:
LGTM, but let's also add a test that this alias exists.
See `tests/queries/0_stateless`

dveeden committed:
Add SCHEMA() to /01705_normalize_case_insensitive_function_names test

dveeden committed:
Add SCHEMA() to test_mysql_protocol

dveeden commented:
> LGTM, but let's also add a test that this alias exists. See `tests/queries/0_stateless`

I've added `SCHEMA()` here. I also added `SCHEMA()` to `tests/integration/test_mysql_protocol/test.py`.
